plugin/src/test/java/org/opensearch/ml/utils/MLNodeUtilsTests.java

```diff
@@ -66,6 +66,31 @@ public void testValidateSchema() throws IOException {
         MLNodeUtils.validateSchema(schema, json);
     }
 
+    @Test
+    public void testValidateEmbeddingInputWithRemoteSchema() throws IOException {
+        String schema = "{\n"
```

Comment:
This unit test ensures this `schema` can work for text embedding. 
I think this is not enough, we should add unit test to cover the three model interfaces changed in this PR.

Reply:
Got it